Mathlib/RingTheory/Polynomial/Resultant/Basic.lean

```diff
@@ -58,12 +59,49 @@ variable (f g : R[X]) (m n : ℕ)
     rw [sylvester, Matrix.of_apply, Fin.addCases_right, Matrix.diagonal_apply]
     split_ifs <;> simp_all [Fin.ext_iff]
 
-end sylvester
+/--
+The Sylvester matrix for `f` and `f.derivative`, modified by dividing the bottom row by
+the leading coefficient of `f`. Important because its determinant is (up to a sign) the
+discriminant of `f`.
+-/
+noncomputable def
+sylvester_deriv (f : R[X]) :
+    Matrix (Fin (f.natDegree - 1 + f.natDegree)) (Fin (f.natDegree - 1 + f.natDegree)) R :=
+  letI n := f.natDegree
+  .of fun i j ↦
+    if i < 2 * n - 2 then
+      f.sylvester f.derivative n (n - 1) ⟨i, by omega⟩ ⟨j, by omega⟩
+    else (if ↑j = n - 2 then 1 else (if ↑j = 2 * n - 2 then n else 0))
+
+/-- We can get the usual Sylvester matrix of `f` and `f.derivative` back from the modified one
+by multiplying the last row by the leading coefficient of `f`. -/
+lemma sylvester_deriv_updateRow (f : R[X]) (hf : 0 < f.natDegree) :
+    (sylvester_deriv f).updateRow ⟨2 * f.natDegree - 2, by omega⟩
+      (f.leadingCoeff • (sylvester_deriv f ⟨2 * f.natDegree - 2, by omega⟩)) =
```

Comment:
No action required here but I wonder if we should define a convenience `Matrix.scaleRow` which performs this operation.